### PR TITLE
Create staging playground deploy workflow and other updates

### LINKED ISSUE #11042: [CI] Continuous Delivery setup to staging-playground.meshery.io
### Current Behavior
Currently staging-playground.meshery.io is deployed manually whenever there are updates.

### Desired Behavior
The [Meshery Build and Releaser (edge)](https://github.com/meshery/meshery/actions/workflows/build-and-release-edge.yml) Github workflow should be updated to automatically deploy staging-playground.meshery.io for each Meshery Edge release

---
### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)
- 🛠 [Meshery Build & Release Strategy](https://docs.meshery.io/project/contributing/build-and-release)
- 📚 [Instructions for contributing to documentation](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#documentation-contribution-flow)
- Meshery documentation [site](https://docs.meshery.io/) and [source](https://github.com/meshery/meshery/tree/master/docs)
- 🎨 Wireframes and designs for Meshery UI in [Figma](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](http://discuss.meshery.io) and [Community Slack](https://slack.meshery.io)
- 🧪 [Meshery Test Plan Spreadsheet](https://docs.google.com/spreadsheets/d/13Ir4gfaKoAX9r8qYjAFFl_U9ntke4X5ndREY1T7bnVs/edit#gid=0)

### PR DESCRIPTION
**Pull request by cpepper96**
**Notes for Reviewers**

This PR fixes #11042 
- Create workflow for updating the meshery helm release in the staging playground. Added this to the edge release workflow
- Updated the playground reset job to use `secrets.METAL_SERVER1`
- Use pinned version of `actions/checkout@v4` in the edge release workflow

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [X] Yes, I signed my commits.

### CODE REVIEW

File: .github/workflows/cncf-staging-playground-deploy-meshery.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+        if: github.repository == 'meshery/meshery'
+        runs-on: ubuntu-latest 
+        steps:
+        - name: Upgrade Meshery Helm Release
+          uses: appleboy/ssh-action@v1.0.3
+          with:
+              host: ${{ secrets.METAL_SERVER3 }}
+              username: root
+              key: ${{ secrets.METAL_SSH_KEY }}
+              script: |
```
Comment (leecalcote) [inline .github/workflows/cncf-staging-playground-deploy-meshery.yaml:17]: It'd be great if we're able to use `mesheryctl system update`.

Comment (leecalcote) [reply .github/workflows/cncf-staging-playground-deploy-meshery.yaml:17]: Not a requirement, but ideal to reuse Meshery's tooling.

**Review by MUzairS15 — COMMENTED** [review]
IMO, using mesheryctl will be ideal and we should start doing that.
It also aligns with the ongoing effort of using Meshery to also provision its remote provider instead of helm.

It also helps identifies where is the pain point or bugs in the flow.
Eventually, we should use mesheryctl for deploying in prod as well.

**Comment by cpepper96** [comment]
Updated to use mesheryctl @leecalcote @MUzairS15

**Comment by leecalcote** [comment]
Good deal. I _think_ that this command is all that we need. We'll find out shortly. 😉

**Review by leecalcote — APPROVED** [review]

**Pull request merged**